mithril-common/src/signable_builder/cardano_transactions.rs

```diff
@@ -89,6 +132,7 @@ impl SignableBuilder<Beacon> for CardanoTransactionsSignableBuilder {
                 .await?;
         }
 
+        let transactions = self.transaction_retriever.get_up_to(&beacon).await?;
```

Comment:
You should add a comment that explain why you need to erase the previous transaction list.

This whole method needs to be refactored I guess. Ideally we should move away the part that update the database so this signable builder could only retrieve the leaf of the "global" merkle tree instead of a list of transactions.

Reply:
Done :heavy_check_mark: